qiskit/algorithms/gradients/lin_comb_estimator_gradient.py

```diff
@@ -91,6 +91,16 @@ def __init__(
         self._derivative_type = derivative_type
         super().__init__(estimator, options)
 
+    @property
```

Comment:
You prefer to replicate these rather than have the getter/setter for this in the base class and perhaps set the actual value from the subclass while defaulting to real in the base class. It does the latter in that it has the getter there which just returns a constant, but maybe if this updateable - and the updating (setter you have here) needs to be public which allows a user to change it too?

Reply:
Since the more general case, which all gradient types support, is only the real derivative I thought it would be safer to add the getters/setters to the special cases which allow changing the `derivative_type`. If we'd put the setter in the base class, most gradient types (paramshift, finite diff, SPSA) would have to disable the setter again, since the only allowed value is `real` 🙂 but I'm happy to change this, too!

Reply:
I don't know - maybe you could just keep the getter in the base class but instead of return a constant it returns a private instance var which defaults to REAL. Then in any derived class where this can be changed just add a setter which sets that private instance var (directly or indirectly via some private method). Maybe that still allows the base class to be changed if someone accesses its private var so maybe you prefer things as they are.

Reply:
Added in [6fac2ec](https://github.com/Qiskit/qiskit-terra/pull/9287/commits/6fac2ec2ad64ace48863db82f6f18caf8297e1b3). The only disadvantage is that pylint now doesn't understand that `self.derivative_type` is a property and not a callable, but that's more a bug on their side 😄